packages/authentication/integration-tests/__tests__/services/module/providers.spec.ts

```diff
@@ -22,6 +23,10 @@ describe("AuthenticationModuleService - AuthProvider", () => {
         schema: process.env.MEDUSA_PRICING_DB_SCHEMA,
       },
     })
+
+    if(service.__hooks?.onApplicationStart) {
+      await service.__hooks.onApplicationStart()
+    }
```

Comment:
todo: I would suggest to start using the medusa app to start modules. you can see how we achieved that in the search pr https://github.com/medusajs/medusa/blob/7de35d84c889070f34dd18255000bb39fba06a72/packages/medusa-test-utils/src/init-modules.ts

Reply:
cc @olivermrbl

Reply:
I am not super happy about these service hooks. The consumer of a module service should never have to worry about the providers being created on load - even when using initialize. 

However, this concern can be discussed and tackled separately from this PR.

Reply:
As we discussed a few times, at some point we need to normalize all entrypoints to be MedusaApp, and discourage the use of `initialize` directly